lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -265,67 +265,207 @@ function _isAccountId(principal) {
     return (principal.length === 12 && /^\d+$/.test(principal));
 }
 
-function _checkPrincipal(requester, principal) {
-    if (principal === '*') {
-        return true;
-    }
-    // User in unauthenticated (anonymous request)
-    if (requester === undefined) {
+function _isRootUser(arn) {
+    if (!arn) {
         return false;
     }
-    if (principal === requester) {
+
+    // Vault returns the following arn when the account makes requests 'arn:aws:iam::295412255825:/master/',
+    // with an empty resource type ('user/' prefix missing).
+    const arns = arn.split(':');
+    const resource = arns.at(-1);
+
+    // If we start with '/' is because we have a empty resource type so we know it is a root account.
+    if (resource.startsWith('/')) {
         return true;
     }
-    if (_isAccountId(principal)) {
-        return _getAccountId(requester) === principal;
+
+    return false;
+}
+
+const checkPrincipalResult = Object.freeze({
+    KO: 0,
+    CROSS_ACCOUNT_OK: 1,
+    OK: 2,
+});
+
+/** _evaluateCrossAccount - checks if it is a cross-account request.
+ * @param {string} requesterARN - requester ARN
+ * @param {string} requesterCanonicalID - requester canonical ID
+ * @param {string} bucketOwnerCanonicalID - bucket owner canonical ID
+ * @return {checkPrincipalResult} OK if it is not cross-account, CROSS_ACCOUNT_OK otherwise.
+ */
+function _checkCrossAccount(requesterARN, requesterCanonicalID, bucketOwnerCanonicalID) {
+    if (!_isRootUser(requesterARN)) {
```

Comment:
for comments consistency:

```suggestion
function _checkCrossAccount(requesterARN, requesterCanonicalID, bucketOwnerCanonicalID) {
   // Vault returns ARNs like 'arn:aws:iam::123456789012:/master/' for root accounts
    // with an empty resource type (missing 'user/' prefix)
    if (!_isRootUser(requesterARN)) {
```